src/quartz/tasograph/tasograph.cpp

```diff
@@ -1583,6 +1600,74 @@ std::shared_ptr<Graph> Graph::optimize(
   return bestGraph;
 } // namespace quartz
 
+std::shared_ptr<Graph> Graph::optimize(std::vector<GraphXfer *> xfers,
+                                       int gate_count_upper_bound,
+                                       std::string circuit_name,
+                                       bool print_message, int timeout) {
+  auto start = std::chrono::steady_clock::now();
+  std::priority_queue<std::shared_ptr<Graph>,
+                      std::vector<std::shared_ptr<Graph>>, GraphCompare>
+      candidates;
+  std::set<size_t> hashmap;
+  std::shared_ptr<Graph> best_graph(new Graph(*this));
+  auto best_gate_cnt = gate_count();
```

Comment:
We want to support other cost functions. It's better to use `total_cost()` here (also change the variable name and the message printed).

Reply:
Done.